server/service/src/requisition/program_settings/customer_program_settings/map.rs

```diff
@@ -0,0 +1,83 @@
+use crate::requisition::program_settings::common::{period_is_available, reduce_and_sort_periods};
+
+use super::{prepare::PrepareProgramSettings, CustomerProgramSettings, OrderType};
+
+/// Map program_settings, order_types, periods and requisitions_in_period to ProgramSettings
+/// based on the customer.
+pub(super) fn map_customer_program_settings(
+    PrepareProgramSettings {
+        settings,
+        order_types,
+        periods,
+        program_customer_and_requisitions_in_periods,
+    }: PrepareProgramSettings,
+) -> Vec<CustomerProgramSettings> {
+    settings
+        .into_iter()
+        .map(|program_setting| {
+            let customer_and_order_type = program_customer_and_requisitions_in_periods
+                .iter()
+                .filter(|(customer, _)| customer.program.id == program_setting.program_row.id)
+                .map(|(customer, requisitions_in_periods)| {
+                    // Filter available periods for program settings for the customer
+                    let order_types = order_types
```

Comment:
Grouping customer with order types to allow different customers the ability to order from same order type if available